python/cuml/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -26,12 +26,160 @@ from cuml.internals.interop import UnsupportedOnGPU, to_cpu, to_gpu
 from cuml.internals.mixins import ClassifierMixin, FMajorInputTagMixin
 from cuml.neighbors.nearest_neighbors import NearestNeighbors
 
-from cython.operator cimport dereference as deref
 from libc.stdint cimport int64_t, uintptr_t
 from libcpp.vector cimport vector
 from pylibraft.common.handle cimport handle_t
 
 
+def _compute_weights(distances, weights):
```

Comment:
This is a fair amount of new python code added to this file (and also in `kneighbors_classifier.pyx`.

Can you instead make a new file (`cuml/neighbors/utils.py`, `cuml/neighbors/weights.py`, :shrug:) to hold all these weight-handling utilities? Moving things that could be pure python out of cython lets all our python tooling work better with them (formatting, linting, ...), and makes them easier to debug and develop.

Reply:
Additionally, it looks like `_compute_weights` is pretty duplicated between the classifier and regressor implementations, it'd be nice to consolidate them.